nautobot_ssot/jobs/base.py

```diff
@@ -210,10 +322,12 @@ def run(self, data, commit):
             self.log_failure(message=f"An exception occurred: `{type(exc).__name__}: {exc}`\n```\n{stacktrace}\n```")
 
 
+# pylint: disable=abstract-method
 class DataSource(DataSyncBaseJob):
     """Base class for Jobs that sync data **from** another data source **to** Nautobot."""
 
     dry_run = BooleanVar(description="Perform a dry-run, making no actual changes to Nautobot data.")
+    memory_profiling = BooleanVar(description="Perform a memory profiling analysis.", default=False)
```

Comment:
Seems like this could be set as-is on the `DataSyncBaseJob` class, without needing to be overridden by the DataSource and DataTarget subclasses, since it doesn't differ between them (unlike `dry_run` which has a different description for each).